Makefile

```diff
@@ -3,12 +3,23 @@ CONTAINER_NAME = webconsoleapp
 SERVER_CONTAINER_NAME = webconsoleserver
 PORT_3SCALE = 8443
 
-build: 3scale/certs/service-chain.pem containers
+build: 3scale/certs/service-chain.pem server/cockpit-bridge-websocket-connector.pyz containers
 
 3scale/certs/service-chain.pem:
 	mkdir -p 3scale/certs && cd 3scale/certs && sscg --subject-alt-name localhost --subject-alt-name host.containers.internal
 	cat 3scale/certs/service.pem 3scale/certs/ca.crt > $@
 
+# bundle https://pypi.org/project/websockets; it's packaged everywhere, but we
+# don't want to install anything on target machines
+# chmod is a hack around https://github.com/python/cpython/issues/96867
+server/cockpit-bridge-websocket-connector.pyz: server/cockpit-bridge-websocket-connector
+	@rm -rf tmp/pyz
+	@mkdir -p tmp/pyz
+	@cp $< tmp/pyz/cockpit_bridge_websocket_connector.py
+	python3 -m pip install --no-compile --target tmp/pyz/ websockets
```

Comment:
Alternatively we could also use the packaged version, and copy

    python3 -c 'import os, websockets; print(os.path.dirname(websockets.__file__))'

However, the pip install is 468 kB, the packaged version is 924 kB. The difference is just `__pycache__`, though (which we could filter out).

Reply:
`python3 -m pip install --no-compile` downloads a binary wheel with the optional speedup extension module and the C file for the extension. I don't see any way to install websockets without the module. I recommend that you `find tmp/pyz/ -name '*.c' -or -name '*.so' -delete`. You could also do some tree shaking and remove WS server related files. `zipapp` also has an option to compress the file.

Reply:
Right, I tried `--no-binary=:all:`, but that doesn't do what it suggests. It still builds the C module, but puts the whole module into an entirely different path (which I don't want). I was looking for some config options, but didn't find anything. So I suppose some manual deletion might be in order.

Reply:
`--no-binary=:all:` downloads the source dist and runs its `setup.py`. The `setuptools` tool chain does not have the concept of optional C extensions and websockets' `setup.py` has no option to disable the extension. Manual cleanup it is!

Reply:
Thanks! I cleaned the .c/.so files, and added --compress. That's now under 100 kB -- at that size I don't think it makes much sense to scalpel off various source files?